docs/src/widget.md

```diff
@@ -0,0 +1,137 @@
+# Widgets overview
+
+The `Widget` trait represents components of your UI. Druid includes a set of
+built-in widgets, and you can also write your own. You combine the built-in
+and custom widgets to create a *widget tree*; you will start with some single
+*root widget*, which will (generally) have children, which may themselves have
+children, and so on. `Widget` has a generic paramater `T` that represents
+the [`Data`] handled by that widget. Some widgets (such as layout widgets)
+may be entirely agnostic about what sort of `Data` they encounter, while other
+widgets (such as a slider) may expect a single type (such as `f64`).
+
+> **Note**: For more information on how different parts of your [`Data`] are exposed
+to different widgets, see [`Lens`].
+
+At a high level, druid works like this:
+
+- **event**: an `Event` arrives from the operating system, such as a key press,
+a mouse movement, or a timer firing. This event is delivered to your root
+widget's [`event`] method. This method is provided **mutable** access to your
+application model; this is the only place where your model can change. Depending
+on the type of `Event` and the implementation of your [`event`] method, this
+event is then delivered recursively down the tree until it is handled.
+- **update**: After this call returns, the framework checks to see if the data was mutated.
+  If so, it calls your root widget's [`update`] method, passing in both the new
+  data as well as the previous data. Your widget can then update any internal
```

Comment:
"internal state" could be confusing to someone who hasn't built a widget yet. Not sure the best way to clarify local state and Data-based state. Naively this looks like it contradicts the line in `event`: "this is the only place where your model can change"